spec/02-integration/05-proxy/10-server_tokens_spec.lua

```diff
@@ -0,0 +1,286 @@
+    })
+
+    teardown(helpers.stop_kong)
+
+    it("should be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "inexistent.com"
+        }
+      })
+
+      assert.res_status(200, res)
+      assert.not_equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.not_equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
+
+    it("should not be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "404.com"
+        }
+      })
+
+      assert.res_status(404, res)
+      assert.equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
+
+  end)
+
+  describe("(with latency_tokens = on)", function()
+
+    setup(start {
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      latency_tokens = "on",
+    })
+
+    teardown(helpers.stop_kong)
+
+    it("should be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "inexistent.com"
+        }
+      })
+
+      assert.res_status(200, res)
+      assert.not_equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.not_equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
+
+    it("should not be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "404.com"
+        }
+      })
+
+      assert.res_status(404, res)
+      assert.equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
+
+  end)
+
+  describe("(with latency_tokens = off)", function()
+
+    setup(start {
+      nginx_conf     = "spec/fixtures/custom_nginx.template",
+      latency_tokens = "off",
+    })
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("should not be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "inexistent.com"
+        }
+      })
+
+      assert.res_status(200, res)
+      assert.equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
+
+    it("should not be returned", function()
+      local res = assert(client:send {
+        method  = "GET",
+        path    = "/get",
+        headers = {
+          host  = "404.com"
+        }
+      })
+
+      assert.res_status(404, res)
+      assert.equal(nil, res.headers[constants.HEADERS.UPSTREAM_LATENCY])
+      assert.equal(nil, res.headers[constants.HEADERS.PROXY_LATENCY])
+    end)
```

Comment:
fyi, you could have done
```
assert.response(res).has.status(404)
assert.response(res).has.header(constants.HEADERS.UPSTREAM_LATENCY)
assert.response(res).has.header(constants.HEADERS.PROXY_LATENCY)
```

tests exactly the same, but provides more context when they fail.